src/frontend/komponenter/ManuellJournalføring/ManuellJournalføring.tsx

```diff
@@ -53,163 +68,183 @@ const ManuellJournalføringContent: React.FC = () => {
+                                modal={{
+                                    className: 'søknad-modal',
+                                    tittel: 'Ønsker du å journalføre uten å knytte til behandling?',
+                                    lukkKnapp: false,
+                                    visModal: visModal,
+                                    actions: [
+                                        <Knapp
+                                            key={'ja'}
+                                            type={'hoved'}
+                                            mini={true}
+                                            spinner={senderInn}
+                                            disabled={senderInn}
+                                            onClick={() => {
+                                                settVisModal(false);
+                                                manueltJournalfør();
+                                            }}
+                                            children={'Ja, journalfør'}
+                                        />,
+                                        <Knapp
+                                            key={'nei'}
+                                            mini={true}
+                                            onClick={() => {
+                                                settVisModal(false);
+                                            }}
+                                            children={
+                                                behandlinger && behandlinger.length > 0
+                                                    ? hentAktivBehandlingForJournalføring()
+                                                        ? 'Nei, velg behandling'
+                                                        : 'Nei, velg/opprett behandling'
+                                                    : 'Nei, opprett behandling'
+                                            }
+                                        />,
+                                    ],
+                                }}
+                            >
+                                <Normaltekst className={'søknad-modal__fjern-vilkår-advarsel'}>
+                                    Du har valgt å journalføre uten å knytte dokumentet til en
+                                    spesifikk behandling. Journalposten knyttes kun til personen.
+                                    <br />
+                                    (Tilsvarende "Knytt til generell sak" i Gosys).
+                                </Normaltekst>
+                            </UIModalWrapper>
+                        )}
+                    </Skjemasteg>
+                    {visDokument && dokumentData.status === RessursStatus.SUKSESS && (
+                        <Bakgrunn>
+                            <iframe
+                                title={'dokument'}
+                                src={dokumentData.data}
```

Comment:
Have you tried setting the URL here instead of fetching the data manually? URLs are supported as long as the respons is a correct pdf.

Reply:
Yes, probably we can use the url. But then we have to move some logic from context to the component, for example, generating the dynamic URL according to the selected document, and error handling. We think the current solution exposes less logic in the UI layer and is in line with the implementation of other features.